doc/user/content/transform-data/optimization.md

```diff
@@ -216,14 +229,70 @@ In many relational databases, indexes don't replicate the entire collection of d
+       Join on=(#0 = #4 AND #5 = #7) type=delta                                    +
+         ArrangeBy keys=[[#0]]                                                     +
+           ReadExistingIndex materialize.public.teachers lookup_value=("Escalante")+
+         ArrangeBy keys=[[#1], [#2]]                                               +
+           Get materialize.public.sections                                         +
+         ArrangeBy keys=[[#0]]                                                     +
+           Get materialize.public.courses                                          +
+                                                                                   +
+ Used Indexes:                                                                     +
+   - materialize.public.teachers_name (lookup)                                     +
+   - materialize.public.sections_fk_teachers (delta join lookup)                   +
+   - materialize.public.pk_courses (delta join lookup)                             +
+   - materialize.public.sections_fk_courses (delta join lookup)                    +
+```
+
+We can see in the above `EXPLAIN` printout that the system will use `teachers_name` for a point lookup, and use three other indexes for the execution of the delta join. Note that the `pk_teachers` index is not used, as explained [above](#joins-with-filters).
+
+The following are the possible index usage types:
+- `*** full scan ***`: We will read the entire index.
+- `lookup`: We will look up only specific keys in the index.
+- `differential join`: We will use the index to perform a so-called _differential join_. For a differential join _between two relations_, the amount of memory required is proportional to the sum of the sizes of each of the input relations _that are not indexed_. In other words, if an input is already indexed, then the size of that input won't affect the memory usage of a differential join between two relations. For a join between more than two relations, we recommend aiming for a delta join instead of a differential join, as explained [above](#optimize-multi-way-joins-with-delta-joins). This is because a differential join between more than two relations will perform a series of binary differential joins on top of each other. Each of these binary joins except the first one will use memory proportional to the size of the intermediate data that is fed into the join.
+- `delta join 1st input (full scan)`: We will use the index for the first input of a [delta join](#optimize-multi-way-joins-with-delta-joins). Note that the first input of a delta join is always fully scanned. However, executing the join won't require additional memory if the input is indexed.
+- `delta join lookup`: We will use the index for a non-first input of a [delta join](#optimize-multi-way-joins-with-delta-joins). This means that in an ad hoc query, the join will perform only lookups into the index.
```

Comment:
```suggestion
The following are the possible index usage types:
- `*** full scan ***`: Materialize will read the entire index.
- `lookup`: Materialize will look up only specific keys in the index.
- `differential join`: Materialize will use the index to perform a _differential join_. For a differential join between two relations, the amount of memory required is proportional to the sum of the sizes of each of the input relations that are **not** indexed. In other words, if an input is already indexed, then the size of that input won't affect the memory usage of a differential join between two relations. For a join between more than two relations, we recommend aiming for a delta join instead of a differential join, as explained [above](#optimize-multi-way-joins-with-delta-joins). A differential join between more than two relations will perform a series of binary differential joins on top of each other, and wach of these binary joins (except the first one) will use memory proportional to the size of the intermediate data that is fed into the join.
- `delta join 1st input (full scan)`: Materialize will use the index for the first input of a [delta join](#optimize-multi-way-joins-with-delta-joins). Note that the first input of a delta join is always fully scanned. However, executing the join won't require additional memory if the input is indexed.
- `delta join lookup`: Materialize will use the index for a non-first input of a [delta join](#optimize-multi-way-joins-with-delta-joins). This means that, in an ad hoc query, the join will perform only lookups into the index.
```